packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -97,31 +102,78 @@ class InMemoryAnchorService extends AnchorService {
     }
 
     for (const compositeKey of Object.keys(validCandidates)) {
-      const candidates: Candidate[] = validCandidates[compositeKey]
-
-      let nonce = 0
-      let selected: Candidate = null
-
-      for (const candidate of candidates) {
-        const record = (await this._ceramic.ipfs.dag.get(candidate.cid)).value
-
-        let currentNonce
-        if (DoctypeUtils.isSignedRecord(record)) {
-          const payload = (await this._ceramic.ipfs.dag.get(record.link)).value
-          currentNonce = payload.header?.nonce || 0
-        } else {
-          currentNonce = record.header?.nonce || 0
+      let candidates: Candidate[] = validCandidates[compositeKey]
+      candidates = await Promise.all(candidates.map(async c => {
+        c.partialLog.push(...await this._loadCommitHistory(c.cid))
+        return c
+      }))
+      candidates.sort((c1, c2) => c2.timestamp - c1.timestamp)
```

Comment:
Yes, we can use timestamps in the **in-memory** anchor service as a heuristic as it is stated in the comments. The `in-memory` anchor service is a mock anchor service used for testing only and it is not meant to be used in production.

Since it is an in-memory anchor service and the data is mocked (`SAMPLE_ETH_TX_HASH`, etc) the timestamps and log check will do the trick until the issue [221](https://github.com/ceramicnetwork/ceramic-anchor-service/issues/221) is resolved.

Feel free to introduce the changes planned for the [221](https://github.com/ceramicnetwork/ceramic-anchor-service/issues/221) into the in-memory service to reflect the real behavior.

Reply:
Issue 221 is for changes to the ceramic-anchor-service repo.  Nothing we're doing here should be dependent on that work.  We want the in-memory anchor service to mirror a real anchor service as much as possible, so while we're in here setting the conflict resolution for js-ceramic, we should make the in-memory anchor service reflect the conflict resolution strategy we are moving towards.  Having the in-memory anchor service have this special behavior then leaks into our tests and affects our ability to test conflict resolution properly

Reply:
timestamps are used just as heuristic, the sort can be removed and the behavior is the same

Reply:
timestamps removed